File: Sample21/UnitTest1.cs
using NUnit.Framework;

namespace Sample21
{
  public class Tests
  {
    [SetUp]
    public void Setup()
    {
    }

    [Test]
    public void Test1()
    {
      var t1 = new Traditional() as ITraditional;
#if NET5_0
      var m1 = new Modern1() as IModern;
      var m2 = new Modern2() as IModern;
#endif
      Assert.That(t1.DoSomething(), Is.EqualTo("Sample21.Traditional"));
#if NET5_0
      Assert.That(m1.DoSomething(), Is.EqualTo("Sample21.Modern1"));
      Assert.That(m2.DoSomething(), Is.EqualTo("** Sample21.Modern2"));
#endif
    }
  }

  public interface ITraditional
  {
    string DoSomething();
  }

  public class Traditional : ITraditional
  {
    public string DoSomething()
    {
      return this.ToString();
    }
  }

#if NET5_0

  public interface IModern
  {
    public string DoSomething()
    {
      return this.ToString();
    }
  }

  public class Modern1 : IModern
  {
  }

  public class Modern2 : IModern
  {
    public string DoSomething()
    {
      return "** " + this.ToString();
    }
  }

#endif
}

File: Sample22/UnitTest1.fs
module Sample22

open System.Xml.Linq
open NUnit.Framework
open Tests

[<SetUp>]
let Setup () =
    ()

let rec RecursiveValidateOpenCover result expected' depth zero expectSkipped =
  let x name = XName.Get(name)
  let rcount = result |> Seq.length

  let expected =
    expected'
    |> Seq.filter (fun (el : XElement) ->
          el.Name.LocalName <> "Module" || expectSkipped || "skippedDueTo"
                                                            |> x
                                                            |> el.Attributes
                                                            |> Seq.isEmpty)
    |> Seq.toList

  let ecount = expected |> Seq.length
  test' <@ rcount = ecount @>
    ("Mismatch at depth " + depth.ToString() + " : " + expected.ToString() + " but got"
      + (result |> Seq.toList).ToString())
  Seq.zip result expected
  |> Seq.iter
        (fun (r : XElement, e : XElement) ->
        test <@ r.Name = e.Name @>
        let ra = r.Attributes()
        let ea = e.Attributes()
        Seq.zip ra ea
        |> Seq.iter
            (fun (a1 : XAttribute, a2 : XAttribute) ->
            test <@ a1.Name = a2.Name @>
            match a1.Name.ToString() with
            | "bev"
            | "visited"
            | "visitedSequencePoints"
            | "visitedBranchPoints"
            | "visitedClasses"
            | "visitedMethods"
            | "sequenceCoverage"
            | "branchCoverage"
            | "uspid"
            | "minCrapScore"
            | "maxCrapScore"
            | "crapScore"
            | "hash" -> ()
            | "fullPath" ->
              test'
                <@ a1.Value.Replace("\\", "/").Replace("altcover", "AltCover").
                            EndsWith(a2.Value.Replace("\\", "/").Replace("altcover", "AltCover")) @>
                (a1.Name.ToString() + " : " + r.ToString() + " -> document")
            | "vc" ->
              let expected =
                maybe zero "0" a2.Value
              test' <@ expected = a1.Value @> (r.ToString() + " -> visitcount")
            | _ ->
              test' <@ a1.Value = a2.Value @>
                (r.ToString() + " -> " + a1.Name.ToString()))
        RecursiveValidateOpenCover (r.Elements()) (e.Elements()) (depth + 1) zero
          expectSkipped)

[<Test>]
let Test1 () =
    Assert.Pass()